src/mds/MDLog.cc

```diff
@@ -99,7 +99,12 @@ class C_MDL_WriteError : public MDSIOContextBase {
       mds->respawn();
     } else {
       derr << "unhandled error " << cpp_strerror(r) << ", shutting down..." << dendl;
-      mds->suicide();
+      // Although it's possible that this could be something transient,
+      // it's severe and scary, so disable this rank until an administrator
+      // intervenes.
+      mds->clog->error() << "Unhandled journal write error on MDS rank " <<
+        mds->get_nodeid() << ": " << cpp_strerror(r) << ", shutting down.";
+      mds->damaged();
```

Comment:
Everywhere else damaged() is invoked, it's followed by an assert(0) or similar. Follow the pattern? (Or maybe phase it out, but you followed it immediately below in the same patch.)